locale/pt-br/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,488 @@
+
+### Que código é executado no Event Loop?
+Quando elas começam, aplicações Node primeiro concluem uma fase de inicialização, fazendo "`require`'ing" de módulos e registrando callbacks para events.
+As Aplicações Node entram no Event Loop, respondendo requisições recebidas do cliente para executar o callback apropriado.
+Esse callback executa de forma síncrona, e pode registrar requisições assíncronas para continuar o processamento após a conclusão.
+
+Os callbacks para essas requisições assíncronas também serão executadas no Event Loop.
+
+O Event Loop também atenderá às requisições assíncronas não-bloqueantes feitas por seus callbacks, por exemplo, I/O de rede.
+
+Em resumo, o Event Loop executa as callbacks JavaScript registradas por eventos, e também é responsável atender requisições assíncronas não-bloqueantes, como I/O de rede.
+
+### Que código é executado na Worker Pool?
+A Worker Pool do Node é implementado na libuv ([docs](http://docs.libuv.org/en/v1.x/threadpool.html)), que expõe uma API geral para envio de tarefas.
+
+O Node usa a Worker Pool para lidar com tarefas "custosas".
+Isso inclui I/O para quais um sistem operacional não fornece uma versão não-bloqueante, bem como tarefas particularmente intensivas em CPU.
+
+Estas são os módulos de APIs do Node que fazem uso dessa Worker Pool:
+
+1. I/O intensivo
+    1. [DNS](https://nodejs.org/api/dns.html): `dns.lookup()`, `dns.lookupService()`.
+    2. [Sistema de arquivo](https://nodejs.org/api/fs.html#fs_threadpool_usage): Todas APIs do sistema de arquivo exceto `fs.FSWatcher()` e aquelas que são explicitamente síncronas usam a threadpool da libuv.
+2. CPU intensivo
+    1. [Crypto](https://nodejs.org/api/crypto.html): `crypto.pbkdf2()`, `crypto.scrypt()`, `crypto.randomBytes()`, `crypto.randomFill()`, `crypto.generateKeyPair()`.
+    2. [Zlib](https://nodejs.org/api/zlib.html#zlib_threadpool_usage): Todas APIs do zlib exceto aquelas que são explicitamente síncronas usam a threadpool da libuv.
+
+Em muitas aplicações Node, essas APIs são as únicas fontes de tarefas para a Worker Pool. Aplicações e módulos que usam um [C++ add-on](https://nodejs.org/api/addons.html) podem enviar tarefas para a Worker Pool.
+
+Para cobrir todos os aspectos, observamos que quando você chama uma dessas APIs a partir de um callback no Event Loop, o Event Loop paga alguns custos menores de configuração, pois entra nas ligações do Node C++ para essa API e envia uma tarefa para ao Worker Pool.
+Esses custos são insignificantes em comparação ao custo total da tarefa, e é por isso que o Event Loop está sendo menos usado.
+Ao enviar uma dessas tarefas para a Worker Pool, o Node fornece um ponteiro para a função C++ correspondente nas ligações Node C++.
+
+### Como o Node decide qual código executar a seguir?
+De forma abstrata, o Event Loop e a Worker Pool mantêm filas para eventos e tarefas pendentes, respectivamente.
+
+Na verdade, o Event Loop não mantém realmente uma fila.
```

Comment:
```suggestion
**Nota**: Na verdade, o Event Loop não mantém realmente uma fila.
```

Reply:
Poderia explicar esse sugestão?

Reply:
Entendi, achei interessante mas precisamos  da opinião de mais pessoas, porque estariamos adicionando algo que não existe no original.

cc @khaosdoctor

Reply:
Let's talk in English in the official repository please :)

Reply:
Por quê?

Reply:
Because it is a public repository, not many people in the world understands portuguese. We're already discussing in English most topics in the official PRs, the only repo where we're talking portuguese is the issue where the translations are requested. Therefore I think it is a good practice to keep all "official" documentation, of code, PRs and such in the original common language

Reply:
And, about the PR, in my opinion, I think adding this won't change much of the meaning but keeps a little more organised, so it may be a good addition

Reply:
ok @khaosdoctor

Reply:
This line is actually part of a paragraph, it makes no sense to put **Nota** at the beginning of this text.

Reply:
Indeed... So I take off my opinion, it is best to translate without `Notas`

Reply:
The upstream "remark-lint" config actually flags instances of "Note that" in the text, but that hasn't been cleaned up in this repo yet